src/amo/components/Collection/index.js

```diff
@@ -111,12 +115,11 @@ export class CollectionBase extends React.Component<Props> {
       collectionChanged = true;
     }
 
-    // When switching into edit mode, refresh the collection add-ons.
-    if (this.props.editing && !nextProps) {
-      addonsPageChanged = true;
-    }
+    const collectionHasAddons = (collection
+      && collection.addons
+      && collection.addons.length > 0);
 
-    if (!collection || collectionChanged) {
+    if (!collection || !collectionHasAddons || collectionChanged) {
```

Comment:
I'm not sure I follow why we needed to add the check for `!collectionHasAddons` here.

Reply:
what i believe i was seeing was that  sometimes there was a collection but the addons hadn't loaded yet..